Shopify/android-testify
# Fix issue #127 - Check for emulator before running Testify tasks

### What this does

Resolves #127

Most Testify plugin tasks would fail with uninformative, confusing errors if an emulator was not found.

e.g.
```
* What went wrong:
Execution failed for task ':Sample:testifyKey'.
> For input string: ""
```

This was common when the emulator was frozen, `adb` had crashed or somehow otherwise was unable to communicate with the emulator. This change introduces an informative error message to indicate that Testify could not find an emulator.

### How to test

- Run `./gradlew Sample:testifyKey`
- Result: 
```
Execution failed for task ':Sample:testifyKey'.
> No Android Virtual Device found. Please start an emulator prior to running Testify tasks.
```

All tasks should fail similarly with no emulator present. When one or more emulators is loaded, all tasks should work normally.
